draft-ietf-tls-tls13.md

```diff
@@ -1854,14 +1820,6 @@ subsequent connections.
 The "key_share" extension contains the endpoint's cryptographic parameters
 for non-PSK key establishment methods (currently DHE or ECDHE).
 
-Clients which offer one or more (EC)DHE cipher suites MUST send this
-extension and SHOULD send at least one supported KeyShareEntry value.
-Servers MUST NOT negotiate any of these cipher suites unless a supported
-value was provided.
-If this extension is not provided in a ServerHello or ClientHello,
```

Comment:
On Thu, Aug 11, 2016 at 11:23 PM, Martin Thomson <notifications@github.com>
wrote:

> In draft-ietf-tls-tls13.md
> <https://github.com/tlswg/tls13-spec/pull/559#discussion_r74546042>:
>
> > @@ -1854,14 +1820,6 @@ subsequent connections.
> >  The "key_share" extension contains the endpoint's cryptographic parameters
> >  for non-PSK key establishment methods (currently DHE or ECDHE).
> >
> > -Clients which offer one or more (EC)DHE cipher suites MUST send this
> > -extension and SHOULD send at least one supported KeyShareEntry value.
> > -Servers MUST NOT negotiate any of these cipher suites unless a supported
> > -value was provided.
> > -If this extension is not provided in a ServerHello or ClientHello,
>
> How about this:
>
> struct {
>   select (role) {
>     case server:
>       bool signed;
>     case client:
>   };
>   SignatureScheme supported_schemes<0..2^16-2>;
> } SignatureAlgorithms;
>
>
I'm not sure I understand the semantics of this proposal. I assume
supported_schemes is
"this is what I would accept".  "signed" is "I, server, am going to sign"?
But then I don't understand
"AuthenticationMode" below.

>  I don't think that it makes sense to put the actual scheme in here. I
> would like to have the supported schemes available for those doing client
> auth; it's neatly symmetrical that way and lets us use this properly in the
> spontaneous case without having to invent a parallel structure.
>
Well, in both cases you have to say "I would accept spontaneous auth",
right?


The only downside is that you have to save this stuff in case you have a
> CertificateVerify.
>
> If you care about that last point, you could make the bool a mode switch:
>
> enum {
>   no_signature, server_signature, both_signature
> } AuthenticationMode;
>
> Which would let you signal that there is a CertificateVerify on the way.
>

You lost me here.

> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/tlswg/tls13-spec/pull/559/files/15126cf5a08c445aeed97c0c25c4f10c2c1b8f26#r74546042>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABD1oXJscb-aF-hDTyuEbQ-w2VpqAgSUks5qfBFTgaJpZM4JUEcw>
> .
>